File: most_digits/solution_3.cs
using System;

public class Kata
{
  public static int FindLongest(int[] number)
  {
    int digits;
    int max_digits = 0;
    int max_index = 0;
    
    for (int i = 0; i < number.Length; i++) {
      digits = (int) Math.Log10(Math.Abs(number[i])) + 1;
      if (digits > max_digits) {
        max_digits = digits;
        max_index = i;
      }
    }
    
    return number[max_index];
  }
}

File: solutions/string_ends_with/solution_3.go
package kata

func solution(str, end string) bool {
  if len(str) == 0 && len(end) > 0 {
    return false
  }
  
  for i, j := len(str)-1, len(end)-1; i >= len(str) - len(end); i, j = i-1, j-1 {
    if string(str[i]) != string(end[j]) {
      return false
    }
  }
  
  return true
}